npm-identities/src/main/java/org/commonjava/atlas/npm/ident/ref/NpmPackageRef.java

```diff
@@ -24,21 +24,22 @@
 import java.io.ObjectOutput;
 
 /**
- * ArtifactRef use jsemver Version object. Ref https://github.com/zafarkhaja/jsemver
+ * NpmPackageRef use jsemver Version object. Ref https://github.com/zafarkhaja/jsemver
  *
  * Created by ruhan on 10/17/18.
  */
-public class NpmArtifactRef implements Externalizable
+public class NpmPackageRef
```

Comment:
It seems like in other places we talk about packages in the abstract, free from versions...where 'package' is almost used the same way as 'project' or 'module' in other languages. We're using it to describe a versioned thing here...if the package is a versioned release, what's the name we should use to describe the stream of packages? Do we need to model that?

For instance, on the Maven side, we actually have ProjectRef for the stream of versioned things, ProjectVersionRef for the versioned release of a project, and ArtifactRef to describe one concrete file in a versioned release. I understand that NPM packages don't have artifacts, per se...or at least, they only have one per versioned release. But do we need to be able to talk about 'my-widget' separately from 'my-widget-1.0.0'?

Reply:
I looked at https://docs.npmjs.com/getting-started/packages and it seems an NPM "package" naturally contains the duple (name, version). So, if we want to only talk about the package's "name", we may need to coin another phrase for it. Maybe "NpmProjectRef"?

Reply:
+1
This is exactly what came into my mind too :-)

Reply:
Add NpmProjectRef. Please review again.